### ISSUE TITLE
When trying to add a filefield, a string value comes in instead of a file

### ISSUE THREAD

**Issue by timuralchin** [aminalaee/sqladmin#357]
### Checklist

- [X] The bug is reproducible against the latest release or `master`.
- [X] There are no similar issues or pull requests to fix it yet.

### Describe the bug

I want to add a file download. But if I add a file widget, it comes with a string instead of a file. How can this be fixed?
```python
class IconAdmin(ModelView, model=Icon):
    form_columns = [Icon.name, Icon.file]
    form_overrides = dict(file=FileField)
```
The file field in the icon model has a string type. Could this be the problem? And if so, what needs to be fixed? 

### Environment

python 3.9.13
sqladmin 0.4.0
sqlmodel 0.0.6

**Comment by timuralchin** [comment]
I solved this one by 
adding custom template with enctype="multipart/form-data" in form

**Comment by aminalaee** [comment]
Can you maybe add the full example solution for future readers?

**Comment by timuralchin** [comment]
So, i have an Icon model:

```python
class Icon(SQLModel, table=True):
    """Icon model."""
    
    name: str
    file: Optional[str] = Field(default=None)
```
and admin model for icon
```python
class IconAdmin(BaseModelView, model=Icon):
    """Icon admin model."""

    create_template = 'icons/create.html'
    edit_template = 'icons/edit.html'
    details_template = 'icons/details.html'
    form_columns = [Icon.name, Icon.file]
    form_overrides = {'file': FileField}

    async def save_file(self, data: Dict, file: UploadFile):
        """Save file to media folder."""
        async with aiofiles.open(
            f'{settings.MEDIA_PATH}/{file.filename}',
            'wb+',
        ) as out_file:
            content = await file.read()
            await out_file.write(content)
            data['file'] = file.filename

    async def insert_model(self, data: dict) -> None:
        """Override insert model method."""
        file: UploadFile = data.pop('file', None)
        if file.filename:
            await self.save_file(data, file)
        return await super().insert_model(data)

    async def update_model(self, pk, data) -> None:
        """Override update model method."""
        file: UploadFile = data.pop('file', None)
        if file.filename:
            await self.save_file(data, file)
        return await super().update_model(pk, data)
```

I couldn't load pics because the html form didn't have the right attribute - I just got the name of the file.
As a result, I just copied the templates into a separate folder and added the right attribute to them. 
Before:
```html
<form method="POST">
```
After
```html
<form method="POST" enctype="multipart/form-data">
```
And then I just specified the path to them in the model.

**Comment by lefiofan** [comment]
Thank you, you helped me a lot.
Please tell me how to make it so that during the editing of the record, the name of the already uploaded file is inserted into the picture field.

Now after creating a record, it shows the name of the picture, but while editing the picture field is empty, after saving the field is empty and overwritten

**Comment by timuralchin** [comment]
@lefiofan 
Since this is the `input` tag type is `file` , the answer is no way.
But you can just pass in the context, for example, `file`,  and render a picture in the template.
To do this, you need to override the `edit` method of the admin class by adding the following lines:
```python
if request.method == "GET":
    if getattr(model, 'file', None):
        model.file = {your image url}
        context["file"] = model.file
```
and add something like this in field row to the edit template:
```html
{% if field.name == 'file' %}
<div class="col-sm-4">
    <img class="form-control" src="{{file}}" width="50px" />
    </div>
{% endif %}
```

**Comment by aminalaee** [comment]
Probably SQLAdmin should provide a FileField for this purpose to make it easier to store files on local disk or S3 or something.

**Comment by lefiofan** [comment]
@timuralchin Thank you

**Comment by aminalaee** [comment]
I've worked on a separate package https://github.com/aminalaee/sqlalchemy-fields which adds FileType and ImageType for SQLAlchemy which work out of the box with SQLAdmin.

This MR adds the docs https://github.com/aminalaee/sqladmin/pull/433 and it will be released in `v0.10.0` but it would be great to give it a go now.

Please note `sqlalchemy-fields` is still in early development but the features and fixes added will be backwards-compatible with SQLAdmin.
